app/src/main/java/com/github/onlynotesswent/model/note/Note.kt

```diff
@@ -60,101 +60,87 @@ data class Note(
-       *
-       * @param userId The ID of the user adding the comment.
-       * @param userName The username of the added comment user.
-       * @param content The text content of the comment.
-       * @param commentCollection The current collection of comments.
-       * @return An updated CommentCollection including the new comment.
-       */
-      fun addComment(
-          userId: String,
-          userName: String,
-          content: String,
-          commentCollection: CommentCollection
-      ): CommentCollection {
-        val mutableCommentsList = commentCollection.commentsList.toMutableList()
-        mutableCommentsList.add(
-            0,
-            Comment(
-                generateId(userId, content),
-                userId,
-                userName,
-                content,
-                Timestamp.now(),
-                Timestamp.now()))
-        return CommentCollection(mutableCommentsList.toList())
-      }
-      /**
-       * Edits an existing comment in the CommentCollection by its ID.
-       *
-       * @param commentId The ID of the comment to be edited.
-       * @param content The new content for the comment.
-       * @param commentCollection The current collection of comments.
-       * @return An updated CommentCollection with the modified comment.
-       */
-      fun editComment(
-          commentId: String,
-          content: String,
-          commentCollection: CommentCollection
-      ): CommentCollection {
-        val mutableCommentsList = commentCollection.commentsList.toMutableList()
-        return CommentCollection(
-            mutableCommentsList.map {
-              if (it.commentId == commentId)
-                  Comment(
-                      generateId(it.userId, content),
-                      it.userId,
-                      it.userName,
-                      content,
-                      it.creationDate,
-                      Timestamp.now())
-              else it
-            })
-      }
-      /**
-       * Deletes a comment from the CommentCollection by its ID.
-       *
-       * @param commentId The ID of the comment to be deleted.
-       * @param commentCollection The current collection of comments.
-       * @return An updated CommentCollection excluding the deleted comment.
-       */
-      fun deleteComment(
-          commentId: String,
-          commentCollection: CommentCollection
-      ): CommentCollection {
-        return CommentCollection(
-            commentCollection.commentsList.filter { it.commentId != commentId })
-      }
+    /**
+     * Edits an existing comment in the CommentCollection by its ID.
+     *
+     * @param commentId The ID of the comment to be edited.
+     * @param content The new content for the comment.
+     * @return An updated CommentCollection with the modified comment.
+     */
+    fun editComment(commentId: String, content: String): CommentCollection {
```

Comment:
Here, agaain it could be renamed to editCommentByCid or ById to make it more specific